### PR TITLE
Fix puna boss dead smoke crash

### PR DESCRIPTION
**Pull request by Nemoel-Tomo**
added enchanged explosion effect, max shockwave angle is 24 now

### CODE REVIEW

**Comment by TokyoSU** [comment]
Tested on my branch and it fixed it for sophia too, approved :)

File: TombEngine/Objects/Effects/Boss.cpp [hunk 1]
```diff
@@ -212,6 +212,12 @@ namespace TEN::Effects::Boss
 			{
 				auto pos = Random::GeneratePointInSphere(sphere);
 				SpawnExplosionSmoke(pos);
+
+				TriggerExplosionSparks(
+					item.Pose.Position.x + (GetRandomControl() % 128 - 64 * 2),
```
Comment (TokyoSU) [inline TombEngine/Objects/Effects/Boss.cpp:217]: Do you know why it's GetRandomControl() % 128 - 64 * 2 ? could you change it with Random::GenerateInt/Float ?

File: TombEngine/Objects/Effects/Boss.cpp [hunk 2]
```diff
@@ -222,18 +228,31 @@ namespace TEN::Effects::Boss
...
```
Comment (TokyoSU) [inline TombEngine/Objects/Effects/Boss.cpp:233]: Same here

File: TombEngine/Objects/Effects/Boss.cpp [hunk 3]
```diff
@@ -222,18 +228,31 @@ namespace TEN::Effects::Boss
...
-			auto orient2D = Random::GenerateAngle();
+			auto orient2D = Random::GenerateAngle(ANGLE(-24.0f), ANGLE(24.0f));
 
 			TriggerShockwave(
 				&shockwavePos, 300, BLOCK(0.5f), speed,
 				color.x * UCHAR_MAX, color.y * UCHAR_MAX, color.z * UCHAR_MAX,
 				36, orient2D, 0);
+
+			TriggerExplosionSparks(
+				item.Pose.Position.x + (GetRandomControl() % 128 - 64 * 2),
```
Comment (TokyoSU) [inline TombEngine/Objects/Effects/Boss.cpp:251]: And same here

File: TombEngine/Objects/Effects/Boss.cpp (hunk 1 above)
Comment (Nemoel-Tomo) [reply TombEngine/Objects/Effects/Boss.cpp]: fixed

File: TombEngine/Objects/Effects/Boss.cpp (hunk 2 above)
Comment (Nemoel-Tomo) [reply TombEngine/Objects/Effects/Boss.cpp]: fixed

File: TombEngine/Objects/Effects/Boss.cpp (hunk 3 above)
Comment (Nemoel-Tomo) [reply TombEngine/Objects/Effects/Boss.cpp]: fixed

**Review by Lwmte — APPROVED** [review]

**Pull request merged**